src/main/java/com/rog/EShop/services/UserService.java

```diff
@@ -38,35 +45,93 @@ public UserDto findById(Long id) {
         }
-
         if (!Objects.equals(userRegisterDto.getPassword(), userRegisterDto.getConfirmPassword())) {
             throw new BadRequestException("Password does not match!");
         }
-        String encode = bCryptPasswordEncoder.encode(userRegisterDto.getPassword());
-        user.setPassword(encode);
-        user.setRegisterDate(LocalDateTime.now());
-        user.setAccountNonExpired(true);
-        user.setAccountNonLocked(true);
-        user.setCredentialsNonExpired(true);
-        user.setEnabled(true);
-        user.setAuthorities(List.of(Role.ROLE_USER));
-        User userSaved = userRepository.save(user);
-        return userMapper.toDTO(userSaved);
 
+        ResponseTokenDto responseTokenDto = getTokenFromKeycloak();
+        ResponseEntity<String> response = getResponseFromKeycloak(userRegisterDto, responseTokenDto);
+        if (response.getStatusCode().isError()) {
+            throw new RuntimeException("Error " + response.getStatusCode());
+        }
+
+        ResponseEntity<List<ResponseKeycloakDto>> responseKeycloakDto = getResponseKeycloakDto(userRegisterDto,
+                responseTokenDto);
+
+        List<ResponseKeycloakDto> responseKeycloakDto1 = responseKeycloakDto.getBody();
+        user.setKeycloakId(responseKeycloakDto1.get(0).getId());
+        Long timeStamp = responseKeycloakDto1.get(0).getCreatedTimestamp();
+        user.setRegisterDate(LocalDateTime.ofInstant(Instant.ofEpochMilli(timeStamp), ZoneOffset.UTC));
+        userRepository.save(user);
+        return userMapper.toDTO(user);
 
     }
 
-    @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return userRepository.findByUsername(username)
-                .orElseThrow(() -> new UsernameNotFoundException("The user " + username + " does not exist"));
+    private static ResponseEntity<List<ResponseKeycloakDto>> getResponseKeycloakDto(UserRegisterDto userRegisterDto,
+                                                                                   ResponseTokenDto responseTokenDto) {
+        RestTemplate restTemplate2 = new RestTemplate();
+        String url = keycloakProperties.getHost() + "/admin/realms/ESTORE/users?username={username}&exact=true";
+        HttpHeaders headers2 = new HttpHeaders();
+        headers2.setBearerAuth(responseTokenDto.getAccessToken());
+        Map<String, String> params = new HashMap<>();
+        params.put("username", userRegisterDto.getUsername());
+        HttpEntity request = new HttpEntity(headers2);
+        ResponseEntity<List<ResponseKeycloakDto>> responseKeycloakDto = restTemplate2
+                .exchange(url, HttpMethod.GET, request,
+                        new ParameterizedTypeReference<List<ResponseKeycloakDto>>() {
+                        }, params);
+        return responseKeycloakDto;
+    }
 
+    private static ResponseEntity<String> getResponseFromKeycloak(UserRegisterDto userRegisterDto,
```

Comment:
Insead of input parameter `ResponseTokenDto responseTokenDto` - Use `String accessToken`
Function name `getResponseFromKeycloak()` is also not understandable. Use `createUserInKeycloak(UserRegisterDto userRegisterDto, String accessToken)`.